src/main/java/org/jenkinsci/plugins/p4/workspace/ManualWorkspaceImpl.java

```diff
@@ -172,12 +172,34 @@ else if (client != null && adjustClientName) {
 	}
 
 
-	private ArrayList<String> getChangeView(WorkspaceSpec workspaceSpec) {
+	private ArrayList<String> getChangeView(IOptionsServer connection, WorkspaceSpec workspaceSpec) throws Exception
+    {
 		ArrayList<String> changeView = new ArrayList<>();
 
 		String view = workspaceSpec.getChangeView();
 		if (view != null) {
 			String specString = getExpand().format(view, true);
+			boolean fromFile = false;
+			if (specString.startsWith("@")) {
+				// extract View from file:	JENKINS-69491.
+				fromFile = true;
+				logger.fine("P4: view from file=" + specString);
+				String specPathFull = specString.substring(1).trim();
+				List<IFileSpec> file = FileSpecBuilder.makeFileSpecList(specPathFull);
+				GetFileContentsOptions printOpts = new GetFileContentsOptions();
+				printOpts.setNoHeaderLine(true);
+				InputStream ins = null;
+				try {
+					connection.getFileContents(file, printOpts);
+					specString = IOUtils.toString(ins, "UTF-8");
+				} finally {
+					if (ins != null) {
+						ins.close();
+					}
+				}
+				specString = getExpand().format(specString, true);
+			}
+
 			for (String line : specString.split("\\n")) {
 				changeView.add(line);
```

Comment:
ClientView in the file can have variables which we need to expand. Better expend it before adding to change view

Reply:
no, the variables would be inside the content of the file, not inside the name of the file